src/Samples/SubModel.Core/Program.fs

```diff
@@ -142,38 +271,54 @@ module App =
-    let clockMsg =
-      DateTimeOffset.Now
-      |> Clock.Tick
-      |> CounterWithClock.ClockMsg
-    dispatch <| App.ClockCounter1Msg clockMsg
-    dispatch <| App.ClockCounter2Msg clockMsg
-  )
-  timer.Start()
-
-
-let main window =
-
-  let logger =
-    LoggerConfiguration()
-      .MinimumLevel.Override("Elmish.WPF.Update", Events.LogEventLevel.Verbose)
-      .MinimumLevel.Override("Elmish.WPF.Bindings", Events.LogEventLevel.Verbose)
-      .MinimumLevel.Override("Elmish.WPF.Performance", Events.LogEventLevel.Verbose)
-      .WriteTo.Console()
-      .CreateLogger()
-
-  WpfProgram.mkSimple App.init App.update App.bindings
-  |> WpfProgram.withSubscription (fun _ -> Cmd.ofSub timerTick)
-  |> WpfProgram.withLogger (new SerilogLoggerFactory(logger))
-  |> WpfProgram.startElmishLoop window
+  
+type public MainViewModel(initialModel, dispatch) as this =
+  inherit ViewModelBase<App2.Model, App2.Msg>(initialModel, dispatch)
+
+  new() = MainViewModel(App2.init (), ignore)
+
+  member _.ClockCounter1 = this.subModel((fun m -> m.ClockCounter1 |> ValueSome), (fun _ -> App2.Msg.ClockCounter1Msg), CounterWithClockViewModel)
+  member _.ClockCounter2 = this.subModel((fun m -> m.ClockCounter2 |> ValueSome), (fun _ -> App2.Msg.ClockCounter2Msg), CounterWithClockViewModel)
+
+type public MainViewModel2() as this =
+  inherit ViewModelBase2<App2.Model, App2.Msg>()
+  let clockCounter1Binding = this.subModel((fun m -> m.ClockCounter1), snd, App2.Msg.ClockCounter1Msg, CounterWithClockViewModel2(), nameof this.ClockCounter1)
+  let clockCounter2Binding = this.subModel((fun m -> m.ClockCounter2), snd, App2.Msg.ClockCounter2Msg, CounterWithClockViewModel2(), nameof this.ClockCounter2)
+
+  member _.ClockCounter1 = clockCounter1Binding
+  member _.ClockCounter2 = clockCounter2Binding
+
+module Program =
+  let counterDesignVm = ViewModel.designInstance Counter.init (Counter.bindings ())
+  let clockDesignVm = ViewModel.designInstance (Clock.init ()) (Clock.bindings ())
+  let counterWithClockDesignVm = ViewModel.designInstance (CounterWithClock.init ()) (CounterWithClock.bindings ())
+  let mainDesignVm = ViewModel.designInstance (App2.init ()) (App2.bindings ())
+
+
+  let timerTick dispatch =
+    let timer = new System.Timers.Timer(1000.)
+    timer.Elapsed.Add (fun _ ->
+      let clockMsg =
+        DateTimeOffset.Now
+        |> Clock.Tick
+        |> CounterWithClock.ClockMsg
+      dispatch <| App2.ClockCounter1Msg clockMsg
+      dispatch <| App2.ClockCounter2Msg clockMsg
+    )
+    timer.Start()
+
+
+  let main window =
+
+    let logger =
+      LoggerConfiguration()
+        .MinimumLevel.Override("Elmish.WPF.Update", Events.LogEventLevel.Verbose)
+        .MinimumLevel.Override("Elmish.WPF.Bindings", Events.LogEventLevel.Verbose)
+        .MinimumLevel.Override("Elmish.WPF.Performance", Events.LogEventLevel.Verbose)
+        .WriteTo.Console()
+        .CreateLogger()
+
+    WpfProgram.mkSimple2 App2.init App2.update MainViewModel
```

Comment:
Using new `WpfProgram` that takes a view model factory instead of a list of bindings.